Doppler.ContactPolicies.Api.Test/UpdateContactPoliciesSettingsTest.cs

```diff
@@ -0,0 +1,164 @@
+
+            var client = _factory.WithWebHostBuilder((e) => e.ConfigureTestServices(services =>
+            {
+                services.AddSingleton(contactPoliciesMock.Object);
+            })).CreateClient();
+
+            var request = new HttpRequestMessage(HttpMethod.Put, $"/accounts/{validAccountName}/settings")
+            {
+                Headers = { { "Authorization", $"Bearer {token}" } },
+                Content = new StringContent(contactPoliciesRequestBodyStub, Encoding.UTF8, "application/json")
+            };
+
+            // Act
+            var response = await client.SendAsync(request);
+
+            // Assert
+            Assert.NotNull(response);
+            Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+        }
+
+        [Fact]
+        public async Task
+            UpdateContactPoliciesSettings_Should_ReturnInternalServerError_When_UserWithSameAccountNameIsFoundButDoesNotHavePermissions()
+        {
+            // Arrange
+            const string contactPoliciesRequestBodyStub = CONTACT_POLICIES_SETTINGS_REQUEST_BODY_STUB;
+            const string validAccountName = "test1@test.com";
+            const string token = TOKEN_ACCOUNT_123_TEST1_AT_TEST_DOT_COM_EXPIRE_20330518;
+
+            var fixture = new Fixture();
+            var foundedIdUser = fixture.Create<int>();
+            var contactPoliciesSettings = new ContactPoliciesSettingsDto
+            {
+                AccountName = validAccountName,
+                IntervalInDays = 10,
+                Active = true,
+                EmailsAmountByInterval = 5,
+                ExcludedSubscribersLists = new List<ExcludedSubscribersLists>()
+            };
+            var contactPoliciesSettingDao = contactPoliciesSettings.ToDao();
+
+            // to allow throw exceptions
+            var contactPoliciesRepositoryMock = new Mock<IContactPoliciesSettingsRepository>(MockBehavior.Strict);
+
+            contactPoliciesRepositoryMock.Setup(x => x.GetIdUserByAccountName(validAccountName))
+                .ReturnsAsync(foundedIdUser);
+            contactPoliciesRepositoryMock.Setup(x => x.UpdateContactPoliciesSettingsAsync(foundedIdUser, contactPoliciesSettingDao))
+                .ThrowsAsync(new Exception("This action is not allowed for this user."));
+
+            var contactService = new ContactPoliciesService(contactPoliciesRepositoryMock.Object);
+
+            var client = _factory.WithWebHostBuilder((e) => e.ConfigureTestServices(services =>
+            {
+                services.AddSingleton(contactPoliciesRepositoryMock.Object);
+                services.AddSingleton(contactService);
+            })).CreateClient();
+            var request = new HttpRequestMessage(HttpMethod.Put, $"/accounts/{validAccountName}/settings")
+            {
+                Headers = { { "Authorization", $"Bearer {token}" } },
+                Content = new StringContent(contactPoliciesRequestBodyStub, Encoding.UTF8, "application/json")
```

Comment:
No deberíamos utilizar el mismo que tenemos en el setup de `UpdateContactPoliciesSettingsAsync`?
```suggestion
                Content = new StringContent(JsonSerializer.Serialize(contactPoliciesSettings), Encoding.UTF8, "application/json")
```

Reply:
No leí todo, pero creo que no estoy de acuerdo con este cambio.

Reply:
JC, cometí un error. Cuando puedas lo hablamos

Reply:
Darian, creo que es mejor dejarlo como un string a serializarlo, de esta forma estamos diciendo explícitamente cómo queremos recibir el Input del update

Reply:
> No deberíamos utilizar el mismo que tenemos en el setup de UpdateContactPoliciesSettingsAsync?

Sobre este comentario te quería indicar que, independientemente de serializar, no tienen relación el modelo que enviemos dentro del `Content` con el modelo que se agregó al `Setup` de `UpdateContactPoliciesSettingsAsync`.